Lib/gftools/builder/operations/__init__.py

```diff
@@ -131,9 +143,24 @@ def validate(self):
 
 
 class FontmakeOperationBase(OperationBase):
+    COMPILER_ENV_KEY = "GFTOOLS_COMPILER"
```

Comment:
couldn't we just have a different subclass of `OperationBase`, and switch to that dynamically? then we'd avoid junking up this class with fontc stuff, and I could properly handle the arguments...

Reply:
If you believe it would be cleaner, I would be happy to see it.

Reply:
The only part I'm not sure about is how best to swap the classes in the presence of the CLI flag. Do you have any ideas?

Reply:
Nope, not really. Once a class is loaded in Python it's unpleasant to try to change its base class. This is why I suggested doing things the way that I did ...

Reply:
I'm not following why we'd need to change base, we should have two different classes and pick which one to create based on cli. I don't know the gftools codebase, is there something about it that makes that difficult?

Reply:
If you're not changing base class, then you will need six different classes (one implementation each of `buildTTF`, `buildOTF`, `buildVariable` for fontc and fontbakery). This is *less* elegant than the solution I'm proposing.

Reply:
I think I would dispute the elegance of writing `if self.compiler ==` :)